### PR TITLE
Implemented Evaluations model and api

### LINKED ISSUE #39: Code QuarterClassesEval model
Implement QuarterClassesEval model to match the design scheme of the database

### LINKED ISSUE #63: Code QuarterClassesEval API
Write up code for QuarterClassesEval API

### PR DESCRIPTION
**Pull request by Benson272**
#PR: 
Related Issues: Tag your issue on the right panel of github page after you submit your PR
Author @Benson272 
Reviewers: @YixuanSeanZhou 
Lable: Set lable to backend on the right panel of the current page

Description: Coded up the evaluations model and api

Added Behavior: Could be optional. Just describe it

Changed Behavior: Could be optional. Just describe it

### CODE REVIEW

File: back_end/src/apis/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                class_id=class_id,
+                                instructor=instructor,
+                                recommend_class=recommend_class,
+                                recommend_instructor=recommend_instructor,
+                                study_hours_per_week=study_hours_per_week,
+                                avg_expected_grade=avg_expected_grade,
+                                avg_grade_received=avg_grade_received)
+
+    if status:
+        return jsonify({'reason': 'evaluation created'}), 200
```
Comment (ljz-1109) [inline back_end/src/apis/evaluation.py]: Also return the object in JSON

File: back_end/src/apis/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    else:
+        return jsonify({'reason': 'evaluation existed'}), 300
+
+
+@evaluation_api_bp.route('/get_evaluations', methods=['GET'])
+def get_evaluations():
+    evaluations = Evaluation.get_evaluations()
+    if evaluations:
+        evaluations = list(map(lambda x: x.to_json(), evaluations))
+    else:
```
Comment (ljz-1109) [inline back_end/src/apis/evaluation.py]: this else loop is not necessary

File: back_end/src/apis/evaluation.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    status = Evaluation.update_evaluation(
+                            id=eval_id, class_id=class_id,
+                            instructor=instructor,
+                            recommend_class=recommend_class,
+                            recommend_instructor=recommend_instructor,
+                            study_hours_per_week=study_hours_per_week,
+                            avg_expected_grade=avg_expected_grade,
+                            avg_grade_received=avg_grade_received)
+
+    ret = Evaluation.get_evaluation(eval_id).to_json()
```
Comment (ljz-1109) [inline back_end/src/apis/evaluation.py]: No need to do another query.

File: back_end/src/models/evaluations.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        if recommend_instructor:
+            self.recommend_instructor = recommend_instructor
+        if study_hours_per_week:
+            self.study_hours_per_week = study_hours_per_week
+        if avg_expected_grade:
+            self.avg_expected_grade = avg_expected_grade
+        if avg_grade_received:
+            self.avg_grade_received = avg_grade_received
+        self.save()
+        return True
```
Comment (ljz-1109) [inline back_end/src/models/evaluations.py]: return self as well

File: back_end/src/models/evaluations.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        db.session.add(evaluation)
+        evaluation.save()
+        return True
+
+    @staticmethod
+    def get_evaluations() -> List[Evaluation]:
+        evaluations = Evaluation.query.all()
+        if evaluations:
+            return evaluations
+        else:
```
Comment (ljz-1109) [inline back_end/src/models/evaluations.py]: No need for else

File: back_end/src/models/evaluations.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    def update_evaluation(id: int, class_id: int = None,
+                          instructor: str = None,
+                          recommend_class: float = None,
+                          recommend_instructor: float = None,
+                          study_hours_per_week: float = None,
+                          avg_expected_grade: str = None,
+                          avg_grade_received: str = None) -> bool:
+
+        evaluation = Evaluation.get_evaluation(evaluation_id=id)
+        return evaluation.update_attr(class_id=class_id, instructor=instructor,
```
Comment (ljz-1109) [inline back_end/src/models/evaluations.py:123]: return "self"

**Review by YixuanSeanZhou — APPROVED** [review]
LGTM!

**Pull request merged**